lib/src/rewrite.rs

```diff
@@ -130,6 +130,22 @@ pub fn rebase_commit_with_options(
     new_parents: &[Commit],
     options: &RebaseOptions,
 ) -> Result<RebasedCommit, TreeMergeError> {
+    // If specified, don't create commit where one parent is an ancestor of another.
+    let mut new_parents = new_parents.iter().cloned().collect_vec();
```

Comment:
nit: `.to_vec()`

or something like this will also work:
```rust
let simplified_new_parents;
if .. {
    simplified_new_parents = ...filter().collect_vec();
    new_parents = &simplified_new_parents;
}
```

Reply:
Went with the second suggestion